docs/userguide/mpi_apps.rst

```diff
@@ -0,0 +1,154 @@
+MPI Apps
+========
+
+.. note::
+
+    Parsl support for MPI Apps is being re-engineered.
+    We describe the best practices with today's Parsl.
+    Join our Slack if you want to steer Parsl's future.
+
+MPI applications run multiple copies of a program that complete a single task by
+coordinating using messages passed within or across nodes.
+Starting MPI application requires invoking a "launcher" code (e.g., ``mpiexec``) from one node
+with options that define how the copies of a program should be distributed to others.
+The need to call a launcher from a specific node requires a special configuration of Parsl `Apps <apps.html>`_
+and `Execution <execution.html>`_ environments to run Apps which use MPI.
+
+HTEx and MPI Tasks
+------------------
+
+The :class:`~parsl.executors.high_throughput.executor.HighThroughputExecutor` (HTEx) is the default execution provider
+available through Parsl.
+Parsl Apps which invoke MPI code require a dedicated HTEx configured such that every Parsl app
+will have access to any of the nodes available within a block,
+and Apps that will invoke the MPI launcher with the correct settings.
+
+Configuring the Provider
+++++++++++++++++++++++++
+
+Parsl must be configured to deploy exactly one executor per execution block.
+This part is simple.
+Instead of defining `a launcher <execution.html#launchers>`_ which will
+place an executor on each node in the block, simply use the :class:`~parsl.launchers.SimpleLauncher`.
+
+It is also necessary to specify the desired number of blocks for the executor.
+Parsl cannot determine the number of blocks needed to run a set of MPI Apps,
+so they must bet set explicitly (see `Issue #1647 <https://github.com/Parsl/parsl/issues/1647>`_).
+The easiest route is to set the ``max_blocks`` and ``min_blocks`` of the provider
+to the desired number of blocks.
+
+Configuring the Executor
+++++++++++++++++++++++++
+
+Configure the executor to launch a number of workers equal to the number of MPI tasks per block.
+First set the ``max_workers`` to the number of MPI Apps you will run per block,
+then set ``cores_per_worker=1e-6`` to avoid limiting by cores available to the executor.
```

Comment:
i'm not really clear in my head what setting cores_per_worker is going to do here... you've already said that the head node of the block will only run max_workers simultaneous apps.  what does cores_per_worker then do beyond that?

Reply:
Setting cores_per_worker fixes this